### PR TITLE
BLD: Fix gitpod meson version

### LINKED ISSUE #55214: BUILD: pandas build failing on Gitpod due to Meson
### Installation check

- [X] I have read the [installation guide](https://pandas.pydata.org/pandas-docs/stable/getting_started/install.html#installing-pandas).

### Platform

Linux-6.1.44-060144-generic-x86_64-with-glibc2.31

### Installation Method

Other

### pandas Version

dev

### Python Version

3.10.8

### Installation Logs

Getting this error after my Gitpod env is setup and I try to execute `python make.py clean`
<details>
Traceback (most recent call last):
  File "/workspace/pandas/doc/make.py", line 382, in <module>
    sys.exit(main())
  File "/workspace/pandas/doc/make.py", line 364, in main
    globals()["pandas"] = importlib.import_module("pandas")
  File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/workspace/pandas/pandas/__init__.py", line 49, in <module>
    from pandas.core.api import (
  File "/workspace/pandas/pandas/core/api.py", line 1, in <module>
    from pandas._libs import (
  File "/workspace/pandas/pandas/_libs/__init__.py", line 16, in <module>
    import pandas._libs.pandas_parser  # noqa: E501 # isort: skip # type: ignore[reportUnusedImport]
ModuleNotFoundError: No module named 'pandas._libs.pandas_parser'
</details>

### PR DESCRIPTION
**Pull request by theuerc**
- [X] closes #55214 
- [X] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

The issue is that meson is out of date in gitpod. 

`pip install --upgrade meson`

The above command fixes the issue, so I added it to the build instructions in gitpod.yml. It could be added to the dockerfile instead, but I decided to be cautious. 

I added the update command to both the `init` and `command` sections because gitpod sometimes skips the `init` section and goes straight to the `command` section without installing anything.

With the fix:
```
gitpod@theuerc-pandas-jy3jlfp1dg6:/workspace/pandas$ python
Python 3.10.8 (main, Dec  6 2022, 14:13:21) [GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pandas
+ /usr/local/bin/ninja
[1/1] Generating write_version_file with a custom command
>>> pandas.to_datetime("2022-01-01")
Timestamp('2022-01-01 00:00:00')
>>> 
```

I'm running the tests in gitpod overnight and I'll check the log file in the morning.

#### Update: 

These are the results:

661 failed, 211445 passed, 7990 skipped, 2426 xfailed, 15 xpassed, 405 warnings, 58 errors in 4524.22s (1:15:24)

The failed tests are the ones that interact with either S3 or a nonexistent SQL server, so I think everything works.

### CODE REVIEW

**Review by datapythonista — APPROVED** [review]
I haven't used Gitpod, and I fail to see why it's being more cautious adding it here than in the Dockerfile, but this seems reasonable to me, happy to get it merged.

**Comment by theuerc** [comment]
I don't think I have permissions to merge this even with approval. I'll update it every so often until someone with write access comes along again.

**Comment by mroeschke** [comment]
Thanks for the PR, but this is currently getting overhauled in https://github.com/pandas-dev/pandas/pull/56375 which should also solve your original issue so closing in favor of that PR

**Pull request closed without merging**
